### PR TITLE
Origin/daniela/liblist migration

### PR DESCRIPTION
**Pull request by danicampagna**
## What is the problem/goal being addressed?
Continue with the migration of the functions written in OCaml to F#, in this case those of the `LibList.fs` class

## What is the solution to this problem?
Functions previously in OCaml, migrated to F#

## What are the changes here? How do they solve the problem and what other product impacts do they cause?
*Please include before/after screenshots/gifs if appropriate*

## How are you sure this works/how was this tested?
Tests passed and I added some aditional tests to try different results.

### CODE REVIEW

File: fsharp-backend/src/LibExecution/StdLib/LibList.fs [hunk 1]
```diff
@@ -684,239 +701,258 @@ let fns : List<BuiltInFn> =
...
+            | Some v -> return v
+          }
+        | _ -> incorrectArgs ())
+      sqlSpec = NotYetImplementedTODO
+      previewable = Pure
+      deprecated = NotDeprecated }
+    { name = fn "List" "filterMap" 0
+      parameters =
+        [ Param.make "list" (TList varA) ""
+          Param.makeWithArgs "f" (TFn([ varA ], TBool)) "" [ "val" ] ]
```
Comment (pbiggar) [inline fsharp-backend/src/LibExecution/StdLib/LibList.fs:757]: The `TBool` here should be a `TOption varB`

File: fsharp-backend/src/LibExecution/StdLib/LibList.fs [hunk 2]
```diff
@@ -684,239 +701,258 @@ let fns : List<BuiltInFn> =
...
+      returnType = TList varA
```
Comment (pbiggar) [inline fsharp-backend/src/LibExecution/StdLib/LibList.fs:758]: This should be a `TList varB`

**Review by pbiggar — CHANGES_REQUESTED** [review]
Nice job!

Apart from the small signature change, can you reformat using the `script/format` script and rebase the changes so that we don't have any formatting changes anymore? Thanks!

**Comment by danicampagna** [comment]
I used the `script/format` script and cherry-picked the commits in PR [#3100](https://github.com/darklang/dark/pull/3100)

**Pull request closed without merging**